OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -135,7 +135,7 @@ public ProductionQueue(ActorInitializer init, Actor playerActor, ProductionQueue
 
 		void INotifyCreated.Created(Actor self)
 		{
-			productionTraits = self.TraitsImplementing<Production>().ToArray();
+			productionTraits = self.TraitsImplementing<Production>().Where(p => p.Info.Produces.Contains(Info.Type)).ToArray();
```

Comment:
I'd like to see and explanation and if appropriate discussion about what this is fixing before we merge.

Reply:
Agreed, I just didn't feel like testing this before filing the PR. Pinging @ABrandau and @MustaphaTR.

Reply:
This fix was required to be able to use a conditional Production trait, that was paired along a ConditionOnProduction trait from @MustaphaTR (the goal was to make customizable units) , Mustapha can elaborate on this further.

Reply:
I think this alone doesn't fix much, some changes from https://github.com/OpenRA/OpenRA/pull/15045 should be required too, but i'm not sure. Without these, TD like UI will keep empty queues for disabled production traits. I had this test code on TD, basically similar thing is going on in NG: https://gist.github.com/MustaphaTR/e8439632e8b23ea1d8d6002d99a6a361 As Nolt said, it requires my GrantConditionOnProduction too tho.

Reply:
Does that mean that we can drop this in favour of #15045?

Reply:
If someone can pick that PR up it would be OK i guess, i don't think IceReaper will be back on PRs soon.